aten/src/ATen/native/cuda/IndexKernel.cpp

```diff
@@ -0,0 +1,73 @@
+#include <ATen/native/cuda/IndexKernel.h>
+#include <ATen/native/TensorAdvancedIndexing.h>  // For at::native::index_out
+#include <ATen/ExpandUtils.h>
+#include <ATen/Functions.h>
+#include <ATen/MemoryOverlap.h>
+#include <ATen/NamedTensorUtils.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+static Tensor & masked_select_out_cuda_impl(Tensor & result, const Tensor & self, const Tensor & mask) {
+  NoNamesGuard guard;
+
+  TORCH_CHECK(mask.scalar_type() == ScalarType::Byte || mask.scalar_type() == ScalarType::Bool,
+              "masked_select: expected BoolTensor or ByteTensor for mask");
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(),
+              "masked_select(): self and result must have the same scalar type");
+
+  auto mask_temp = (mask.dim() == 0)
+    ? c10::MaybeOwned<Tensor>::owned(mask.unsqueeze(0))
+    : c10::MaybeOwned<Tensor>::borrowed(mask);
+  auto self_temp = (self.dim() == 0)
+    ? c10::MaybeOwned<Tensor>::owned(self.unsqueeze(0))
+    : c10::MaybeOwned<Tensor>::borrowed(self);
+
+  // Cannot reassign to mask_temp and self_temp here! if they are
+  // owning and expand_outplace returns a borrow, the returned borrow
+  // would dangle.
+  auto mask_self_expanded = expand_outplace(*mask_temp, *self_temp);
+  at::native::index_out(result, *std::get<1>(mask_self_expanded), c10::List<c10::optional<at::Tensor>>({*std::get<0>(std::move(mask_self_expanded))}));
```

Comment:
from internal lint:
> Potential variable access after move.
> Variable mask_self_expanded is accessed here, and is moved-from in an unspecified order

Reply:
i see a formatting improvement, but I don't see any material changes here.

Reply:
The `std::move` now comes after the `std::get`, so only one tuple element is created as an r-value reference.

Reply:
ah thanks, i see that now.